packages/clay-core/src/vertical-bar/Resizer.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * SPDX-FileCopyrightText: © 2023 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {Keys} from '@clayui/shared';
+import React, {useContext} from 'react';
+
+import {VerticalBarContext} from './context';
+
+type Props = {
+	/**
+	 * Sets the CSS className for the component.
+	 */
+	className?: string;
+
+	/**
+	 * Reference to Panel
+	 */
+	nodeRef?: React.RefObject<HTMLDivElement>;
+};
+
+let keyDownCounter = 0;
+
+export function Resizer({nodeRef, ...otherProps}: Props) {
+	const {
+		internalPanelWidth,
+		onPanelWidthChange,
+		panelWidthMax,
+		panelWidthMin,
+		position,
+	} = useContext(VerticalBarContext);
+
+	const positionLeft = position === 'left';
+
+	const getStartWidth = () => {
+		return nodeRef?.current?.offsetWidth || 320;
+	};
+
+	const decreasePanelWidth = (delta = 1, startWidth = getStartWidth()) => {
+		const width =
+			internalPanelWidth - delta < panelWidthMin
+				? panelWidthMin
```

Comment:
I think when the value reaches the minimum value we only need to call it once to avoid calling this function every time.